test/functional/feature_blockfilterindex_prune.py

```diff
@@ -13,43 +13,53 @@
 
 class FeatureBlockfilterindexPruneTest(BitcoinTestFramework):
     def set_test_params(self):
-        self.num_nodes = 2
-        self.extra_args = [["-fastprune", "-prune=1"], ["-fastprune", "-prune=1", "-blockfilterindex=1"]]
+        self.num_nodes = 1
+        self.extra_args = [["-fastprune", "-prune=1", "-blockfilterindex=1"]]
+
+    def sync_index(self, height):
+        expected = {'basic block filter index': {'synced': True, 'best_block_height': height}}
+        self.wait_until(lambda: self.nodes[0].getindexinfo() == expected)
 
     def run_test(self):
         self.log.info("check if we can access a blockfilter when pruning is enabled but no blocks are actually pruned")
-        self.wait_until(lambda: self.nodes[1].getindexinfo() == {'basic block filter index': {'synced': True, 'best_block_height': 200}})
-        assert len(self.nodes[1].getblockfilter(self.nodes[1].getbestblockhash())['filter']) > 0
+        self.sync_index(height=200)
+        assert_greater_than(len(self.nodes[0].getblockfilter(self.nodes[0].getbestblockhash())['filter']), 0)
         # Mine two batches of blocks to avoid hitting NODE_NETWORK_LIMITED_MIN_BLOCKS disconnection
-        self.nodes[1].generate(250)
+        self.nodes[0].generate(250)
         self.sync_all()
-        self.nodes[1].generate(250)
+        self.nodes[0].generate(250)
         self.sync_all()
-        self.wait_until(lambda: self.nodes[1].getindexinfo() == {'basic block filter index': {'synced': True, 'best_block_height': 700}})
+        self.sync_index(height=700)
+
         self.log.info("prune some blocks")
-        pruneheight = self.nodes[1].pruneblockchain(400)
+        pruneheight = self.nodes[0].pruneblockchain(400)
         assert_equal(pruneheight, 250)
+
         self.log.info("check if we can access the tips blockfilter when we have pruned some blocks")
-        assert len(self.nodes[1].getblockfilter(self.nodes[1].getbestblockhash())['filter']) > 0
+        assert_greater_than(len(self.nodes[0].getblockfilter(self.nodes[0].getbestblockhash())['filter']), 0)
+
         self.log.info("check if we can access the blockfilter of a pruned block")
-        assert len(self.nodes[1].getblockfilter(self.nodes[1].getblockhash(2))['filter']) > 0
+        assert_greater_than(len(self.nodes[0].getblockfilter(self.nodes[0].getblockhash(2))['filter']), 0)
+
         self.log.info("start node without blockfilterindex")
-        self.stop_node(1)
-        self.start_node(1, extra_args=self.extra_args[0])
+        self.stop_node(0)
+        self.start_node(0, extra_args=["-fastprune", "-prune=1"])
```

Comment:
```suggestion
        self.restart_node(0, extra_args=["-fastprune", "-prune=1"])
```

might be possible to write shorter

Reply:
nice! done